pkg/agent-functions/agents/agents.go

```diff
@@ -15,34 +21,60 @@ var Module = fx.Provide(NewAgents)
 // Agents wraps rpc.Clients where clients are agents and provides wrapper
 //
 // Agents wraps functions registered in agentfunctions, types should match.
-type Agents struct{ *rpc.Clients }
+type Agents struct {
+	*rpc.Clients
+	etcdTransport *transport.EtcdTransportServer
+	etcdClient    *etcdclient.Client
+}
 
 // NewAgents wraps Clients with Agent-specific function wrappers.
-func NewAgents(clients *rpc.Clients) Agents { return Agents{Clients: clients} }
+func NewAgents(transport *transport.EtcdTransportServer, client *etcdclient.Client) Agents {
+	return Agents{
+		etcdTransport: transport,
+		etcdClient:    client,
+	}
+}
 
 // ListFlowControlPoints lists control points of all agents.
 //
 // Handled by agentfunctions.ControlPointsHandler.
-func (a Agents) ListFlowControlPoints() ([]rpc.Result[*cmdv1.ListFlowControlPointsAgentResponse], error) {
+func (a Agents) ListFlowControlPoints() ([]transport.Result[*cmdv1.ListFlowControlPointsAgentResponse], error) {
```

Comment:
These functions do not take agent group right but they should

Reply:
Should create a list agent groups command as well

Reply:
Will be taking this in a consecutive PR